acquia_lift.drush.inc

```diff
@@ -32,3 +36,64 @@ function drush_acquia_lift_menu_rebuild() {
   _acquia_lift_build_menu('acquia-lift-controls');
   drush_log('The menu has been rebuilt.', 'success');
 }
+
+/**
+ * Syncs the Acquia Lift Campaigns.
+ *
+ * This command is an alternative to the UI's "Sync with Lift" buttons.
+ */
+function drush_acquia_lift_campaign_sync() {
+  module_load_include('inc', 'acquia_lift', 'acquia_lift.batch');
+
+  // Step 1. Exit if there is no campaign to be synchronized.
+  $agents = array();
+  foreach (acquia_lift_get_agent_types() as $type => $info) {
+    $agents = array_merge($agents, personalize_agent_load_by_type($type));
+  }
+  if (empty($agents)) {
+    drush_log(dt('There is no campaign to be synchronized.'), 'success');
+    return;
+  }
+
+  // Step 2. Select sync type.
+  $options = array(
+    'all' => dt('Sync all campaigns.'),
+    'one' => dt('Sync one campaign by its machine name.'),
+  );
+  $type_selection = drush_choice($options, dt('Select sync type:'));
+  if (!$type_selection) {
+    return;
+  }
+
+  // Step 3a. Select a campaign (if the type is "one").
+  if ($type_selection == 'one') {
+    $agent_names = array_keys($agents);
+    $name_selection = drush_choice($agent_names, dt('Select a campaign machine name to sync:'));
+    if (!$name_selection) {
+      return;
+    }
+    $agent_name = $agent_names[$name_selection];
+    $agent = $agents[$agent_name];
+    $agents = array($agent_name => $agent);
+  }
+
+  // Step 4. Sync selected campaign(s)
+  // First delete the queue because we are about to sync everything now.
+  acquia_lift_delete_queue();
+  foreach ($agents as $agent_name => $agent) {
+    $operations = acquia_lift_get_sync_operations_for_agents(array($agent_name => $agent));
+    $batch_operations = array();
+    foreach ($operations as $operation) {
+      $batch_operations[] = array('acquia_lift_batch_process_item', array($operation));
+    }
+    $batch = array(
+      'operations' => $batch_operations,
+      'finished' => 'acquia_lift_batch_finished',
+      'file' => drupal_get_path('module', 'acquia_lift') . '/acquia_lift.batch.inc',
+    );
+    batch_set($batch);
+    $batch =& batch_get();
+    $batch['progressive'] = false;
+    drush_backend_batch_process();
```

Comment:
Hmm, I wonder if we're getting anything at all out of using the batch API here... what do you think? Maybe we should just get the list of operations for each agent, process them directly (i.e. just call the method) and output a message when each agent has had all its components sync'd?